### PR TITLE
Switch IteratorTest to use TF combinations

### PR DESCRIPTION
**Pull request by feihugis**
This PR switch `IteratorTest` to use TF combinations.

### CODE REVIEW

File: tensorflow/python/data/kernel_tests/iterator_test.py [hunk 1]
```diff
@@ -778,34 +799,43 @@ def testRepeatedGetNextWarning(self):
...
-          "b": (ops.Tensor, ops.Tensor)
-      }, {
-          "a": dtypes.float32,
-          "b": (dtypes.string, dtypes.string)
-      }, {
-          "a": [],
-          "b": ([1], [])
-      }),
-  )
+  @combinations.generate(combinations.times(
```
Comment (jsimsa) [inline tensorflow/python/data/kernel_tests/iterator_test.py]: As a rule of thumb, we have agreed on the tf.data team that going forward we should not used parametrized tests that require `lambda` parameters and instead create separate unit tests. Could you please split this tests into separate tests, one for each test case? Thank you.

Comment (feihugis) [reply tensorflow/python/data/kernel_tests/iterator_test.py]: Sure, will do it.

File: tensorflow/python/data/kernel_tests/iterator_test.py [hunk 2]
```diff
@@ -843,7 +913,8 @@ def testIteratorEagerIteration(self, execution_mode):
         self.assertEqual(val, foo.numpy())
         val += 1
 
-  @test_util.run_v2_only
+  @combinations.generate(
+      combinations.combine(tf_api_version=2, mode="eager"))
```
Comment (feihugis) [inline tensorflow/python/data/kernel_tests/iterator_test.py]: I'm not sure if my change here is correct. 
* IIUC, `@test_util.run_v2_only` will run the test in both eager and graph mode, but only the eager tensor has `numpy()`, which is used at [self.assertEqual(queue.size().numpy(), 2)](https://github.com/tensorflow/tensorflow/pull/32599/files#diff-edaa5071d8e3b55304a0797163bb7cbfL861). 
* The change here will only run the test in the eager mode.

**Comment by feihugis** [comment]
@jsimsa `testIteratorStructure()` is split into three test functions to avoid the use of `lambda` parameter. Could you please take a look at the change(https://github.com/tensorflow/tensorflow/pull/32599/commits/7268f8dc5cd53471b129cb52f76c4b3147ffd173)?

File: tensorflow/python/data/kernel_tests/iterator_test.py (hunk 2 above)
Comment (jsimsa) [reply tensorflow/python/data/kernel_tests/iterator_test.py]: `@test_util.run_v2_only` runs V2 * eager
`@test_util.run_v1_only` runs V1 * graph
`@test_util.deprecated_graph_mode_only` runs (V1 + V2) * graph

**Review by jsimsa — APPROVED** [review]

File: tensorflow/python/data/kernel_tests/iterator_test.py (hunk 2 above)
Comment (feihugis) [reply tensorflow/python/data/kernel_tests/iterator_test.py]: Gotcha, thanks!

**Comment by feihugis** [comment]
@gbaned An error happened while migrating the change for the internal check. Could you please help re-trigger the internal check? Thanks!

**Pull request merged**